cmake/std/PullRequestLinuxDriverTest.py

```diff
@@ -0,0 +1,483 @@
+    print("  pwd = {cwd}".format(cwd=os.getcwd()), file=sys.stdout)
+    print("", file=sys.stdout)
+    for key in os.environ:
+        print(key + ' = ' + os.environ[key],
+              file=sys.stdout)
+    print(
+        "\n==========================================================================================",
+        file=sys.stdout)
+    print(module('list'))
+
+
+def getCDashTrack():
+    returnValue = 'Pull Request'
+    if 'PULLREQUEST_CDASH_TRACK' in os.environ:
+        returnValue = os.environ['PULLREQUEST_CDASH_TRACK']
+        print('PULLREQUEST_CDASH_TRACK is set. Setting CDASH_TRACK={}'.format(
+            returnValue) )
+        returnValue = os.environ['PULLREQUEST_CDASH_TRACK']
+    else:
+        print('PULLREQUEST_CDASH_TRACK isn\'t set, using default value')
+
+    return returnValue
+
+def compute_n():
+    '''given the default and the hardware environment determine the
+     number of processors  to use'''
+    try:
+        environment_weight = int(os.environ['JENKINS_JOB_WEIGHT'])
+    except KeyError:
+        environment_weight = 29
+    requested_mem_per_core =  3.0
+
+    n_cpu = cpu_count()
+    # this assumes unix/linux systems. A more general
+    # solution is available in psutil at the cost of
+    # using anaconda - which is not  bad.
+    with open('/proc/meminfo') as f_ptr:
+        meminfo = dict((i.split()[0].rstrip(':'), int(i.split()[1])) for i in
+                        f_ptr.readlines())
+    mem_kib = meminfo['MemTotal']
+    mem_G = mem_kib/(1024*1024)
+    number_possible_jobs = max(1, int(n_cpu/environment_weight))
+    parallel_level = int(mem_G/(requested_mem_per_core*number_possible_jobs))
+
+    if parallel_level > environment_weight:
+        parallel_level = environment_weight
+    return parallel_level
+
+
+config_map = {'Trilinos_pullrequest_gcc_4.8.4': 'PullRequestLinuxGCC4.8.4TestingSettings.cmake',
+              'Trilinos_pullrequest_intel_17.0.1': 'PullRequestLinuxIntelTestingSettings.cmake',
+              'Trilinos_pullrequest_gcc_4.9.3_SERIAL': 'PullRequestLinuxGCC4.9.3TestingSettingsSERIAL.cmake',
+              'Trilinos_pullrequest_gcc_7.2.0': 'PullRequestLinuxGCC7.2.0TestingSettings.cmake',
+              'Trilinos_pullrequest_cuda_9.2': 'PullRequestLinuxCuda9.2TestingSettings.cmake',
+              'Trilinos_pullrequest_python_2': 'PullRequestLinuxPython2.cmake',
+              'Trilinos_pullrequest_python_3': 'PullRequestLinuxPython3.cmake'}
+
+
+def run():
+    return_value = True
+    arguments = parse_args()
+
+    os.chdir(arguments.workspaceDir)
+
+    print_input_variables(arguments)
+
+    re_src_branchname = "master_merge_[0-9]{8}_[0-9]{6}"
+    if "master" == arguments.targetBranch:
+        if re.match(re_src_branchname, arguments.sourceBranch) is None:
+            sys.exit("""------------------------------------------------------------------------------------------
+NOTICE: Destination branch is trilinos/Trilnos::master
```

Comment:
We use the same infrastructure for dev-> master testing. If source branch is develop, this is ok.